std-bits/table/src/main/java/org/enso/table/data/column/operation/CountUntrimmed.java

```diff
@@ -0,0 +1,59 @@
+package org.enso.table.data.column.operation;
+
+import java.util.Random;
+import org.enso.base.Text_Utils;
+import org.enso.table.data.column.storage.ColumnStorage;
+import org.enso.table.data.column.storage.StringStorage;
+import org.enso.table.data.table.Column;
+import org.graalvm.polyglot.Context;
+
+public class CountUntrimmed {
+  // Default seed for random number generation (no specific reason for this value, just stability on
+  // result).
+  private static final long RANDOM_SEED = 677280131;
+
+  // Default sample size for counting untrimmed cells.
+  public static final long DEFAULT_SAMPLE_SIZE = 10000;
+
+  /** Counts the number of cells in the columns with leading or trailing whitespace. */
+  public static long apply(Column column, long sampleSize) {
+    ColumnStorage storage = column.getStorage();
+    return applyToStorage(storage, sampleSize);
+  }
+
+  /** Counts the number of cells in the given storage with leading or trailing whitespace. */
+  public static long applyToStorage(ColumnStorage storage, long sampleSize) {
+    return (sampleSize == DEFAULT_SAMPLE_SIZE && storage instanceof StringStorage stringStorage)
+        ? stringStorage.countUntrimmed()
+        : compute(storage, sampleSize);
+  }
+
+  /** Internal method performing the calculation on a storage. */
+  public static long compute(ColumnStorage storage, long sampleSize) {
+    long size = storage.getSize();
+    boolean sample = sampleSize < size;
+    Random rng = sample ? new Random(RANDOM_SEED) : null;
+    double sampleRate = sample ? (double) sampleSize / size : 1.0;
+
+    Context context = Context.getCurrent();
+    long count = 0;
+    for (long i = 0; i < storage.getSize(); i++) {
+      if (sample && rng.nextDouble() > sampleRate) {
+        continue;
+      }
```

Comment:
It feels like we could be sampling more efficiently?

If `size` is **MUCH** bigger than `sampleSize` (e.g. `size = 100M`), we will still run `size` iterations here. Most of these iterations will be a no-op `continue`, but that still requires a numerous amount of iterations.

Wouldn't it be more efficient to do:
```suggestion
    long count = 0;
    for (long j = 0; j < sampleSize; j++) {
        int i = rng.nextInt(size);
```

i.e. to just iterate over `sampleSize` entries and choose the index to sample randomly each time?

Then regardless of how large `size` is, we do at most `sampleSize` iterations. The behaviour should be roughly equivalent statistically - both algorithms randomly pick `sampleSize` samples from the dataset, just one runs in `O(size)` time and another in `O(sampleSize)` (i.e. essentially constant).

The difference between algorithms is that my suggestion samples with replacement - it _may_ choose the same index multiple times for the sample. But I'm not sure if this is a problem for this statistic?